pkg/instance/service.go

```diff
@@ -869,3 +812,106 @@ func (s service) destroy(instance *model.Instance) error {
 	}
 	return nil
 }
+
+type updateMessage struct {
+	EventType    watch.EventType `json:"eventType"`
+	DeploymentId uint            `json:"deploymentId"`
+	InstanceId   uint            `json:"instanceId"`
+	Status       Status          `json:"status"`
+}
+
+func (s service) ListenForClusterUpdates() {
+	fakeAdministrator := &model.User{
+		Groups: []model.Group{
+			{
+				Name: model.AdministratorGroupName,
+			},
+		},
+	}
+
+	groups, err := s.groupService.FindAll(fakeAdministrator, true)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	for _, group := range groups {
+		g := group
+		go func() {
+			ks, err := NewKubernetesService(g.ClusterConfiguration)
+			if err != nil {
+				log.Printf("unable to get create new kubernetes service: %v", err)
+			}
+
+			for {
+				events, err := ks.watch(g.Name)
+				if err != nil {
+					log.Printf("error watching %q: %v", g.Name, err)
```

Comment:
do you expect the channel to return any events in case of an error? I doubt it so would return